### PR TITLE
Add warnings and fix numba

### PR DESCRIPTION
**Pull request by Adel-Moumen**
## What does this PR do?

This PR adds back flags: `--auto_mix_prec` and `--bfloat16_mix_prec`; however, a warning is raised, suggesting to use `--precision=fp16` (or bf16) for a smooth transition to SB 1.0.

Furthermore, I added back the loss inside the `torch.autocast` region. The reason is that Numba/PyTorch transducer losses require inputs to be in the same dtype. Additionally, one reason to leave it there is that now, the training is much more stable and shouldn't be an issue to wrap everything in the region.

I tried it on fp16, and it works as expected with Numba and PyTorch Transducer loss."

<details>
  <summary><b>Before submitting</b></summary>

- [ ] Did you read the [contributor guideline](https://speechbrain.readthedocs.io/en/latest/contributing.html)?
- [ ] Did you make sure your **PR does only one thing**, instead of bundling different changes together?
- [ ] Did you make sure to **update the documentation** with your changes? (if necessary)
- [ ] Did you write any **new necessary tests**? (not for typos and docs)
- [ ] Did you verify new and **existing [tests](https://github.com/speechbrain/speechbrain/tree/develop/tests) pass** locally with your changes?
- [ ] Did you list all the **breaking changes** introduced by this pull request?
- [ ] Does your code adhere to project-specific code style and conventions?

</details>

## PR review

<details>
  <summary>Reviewer checklist</summary>

- [ ] Is this pull request ready for review? (if not, please submit in draft mode)
- [ ] Check that all items from **Before submitting** are resolved
- [ ] Make sure the title is self-explanatory and the description concisely explains the PR
- [ ] Add labels and milestones (and optionally projects) to the PR so it can be classified
- [ ] Confirm that the changes adhere to compatibility requirements (e.g., Python version, platform)
- [ ] Review the self-review checklist to ensure the code is ready for review

</details>

### CODE REVIEW

**Comment by TParcollet** [comment]
I tested and it works as intended. LGTM.
